[blacksesamezongzi] iP

blacksesamezongzi opened the pull request:
### AdventureGuide
> "Life is a great adventure, or it is nothing."  — John Dewey ([Source](https://www.mymountainsandme.com/blog/adventure-quotes))

AdventureGuide helps you out in this exciting yet sometimes upsetting adventure of life, by recording all the things you need to do at one step. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

Steps to start your journey with AdventureGuide...
1. download jar file from github
2. double click the jar file
3. add your tasks
4. let it manage your tasks for you 😺 

And it is **FREE**!

Features at hand:
- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Rename main class

Commit:
add gitignore

Commit:
Add echo feature in AdventureGuide.java

Commit:
Add storage feature to AdventureGuide

Commit:
Add Task.java, move main code to AdventureGuideBot.java

Commit:
Add deadline, event, todo as subclass of Task

Commit:
Add Exceptions pacage to handle exceptions more easily. Extract actions as methods in adventureGuide.java

Commit:
Add remove feature in AdventureGuideBot.java

YosieSYX commented on src/main/java/components/AdventureGuideBot.java:
```diff
@@ -0,0 +1,154 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import Exceptions.*;
```
Would it be better to expand out explicitly each package that you imported here? So then the users will be able to know exactly what packages they need.

ypuppy replied:
would it fits the coding style better if you dont use wildcard importing ?

YosieSYX commented on src/main/java/components/AdventureGuideBot.java:
```diff
@@ -0,0 +1,154 @@
+    }
+
+    private void handleEvent(String input) throws EmptyDescriptionException {
+        String[] parts = input.substring(6).split(" /from | /to ");
```
Would it be better to give this variable 'parts' a more meaningful name? Because currently it's not too clear for me what exactly this string stores.

YosieSYX commented on src/main/java/components/AdventureGuideBot.java:
```diff
@@ -0,0 +1,154 @@
+    private void handleDelete(String input) throws InvalidTaskNumberException {
+        int taskIndex = Integer.parseInt(input.split(" ")[1]) - 1;
+        if (taskIndex >= 0 && taskIndex < tasks.size()) {
+            Task removedTask = tasks.remove(taskIndex);
```
I personally feel that the variable name 'removedTask is a bit misleading because it is actually the task that is about to be removed but not yet removed. Would a name like: requestedToBeRemovedTask be better?

YosieSYX commented on src/main/java/components/Task.java:
```diff
@@ -0,0 +1,28 @@
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
I like this boolean variable name.

YosieSYX reviewed (commented):
Overall good and clean code. Some minor issues I identified are that some variable names are not meaningful enough. Consider adjusting a bit if that makes sense.

ypuppy commented on src/main/java/components/AdventureGuideBot.java:
```diff
@@ -0,0 +1,154 @@
+        }
+        tasks.add(new Deadline(description, by));
+        System.out.println("____________________________________________________________");
+        System.out.println(" Got it. I've added this task:");
```
would it be better if you separate the print function and input taking function in other files so it cleaner and easier for debugging and make changes later on

ypuppy reviewed (commented):
I like your code, its very neat.

Commit:
Add save and load feature

The user previously had to restate all their tasks once they have exit,
which can be disturbing.

To handle this problem we add follwing features in this stage:
* Store the tasks locally once an update is made
* Load the tasks once the user start the program

Commit:
Update date formate of Deadline

Commit:
Update date format of Event

Commit:
Remove extra import in Event.java

Commit:
Add methods to get dates of different types from Event and Deadline

Commit:
Add gitingore to ingore data/tasks.txt

Commit:
Ignore data/tasks/txt

Commit:
Merge branch 'branch-Level-8'

Commit:
Add Ui.java

Commit:
Add Parser.java, TaskList.java

Commit:
Use Parser, TaskList, ui, Storage in AdtureGuideBot.java

Commit:
Categorize classes with packages

Commit:
Merge branch 'add-gradle-support' of https://github.com/blacksesamezongzi/ip

Commit:
Add Gradle feature, configure build.gradle file

Commit:
Add Testcases for gradlew test

Commit:
Add TaskListTest.java

Commit:
Add comments to describe methods and classes

Commit:
Modify Coding Standard

Commit:
Add find feature

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/components/Parser.java
#	src/main/java/duke/ui/AdventureGuideBot.java

Commit:
Update depencencies

Commit:
Update entry point as Main.java

Commit:
Add MainWindow.java

Commit:
Correct package name of MainWindow

Commit:
Add DialogBox.java

Commit:
Add varargs features in DialogBox.java and MainWindow.java

Commit:
Ignore task.txt

Commit:
Update depencencies of gradle

Commit:
Add UI feature with responses being echo as placeholder

Commit:
Add UI feature to AdventureGuide

Commit:
Add assertions to AdventureGuideBot.java

There are only exception handlers to check exceptions.

Assertions can be added to check assumptions that should hold at
various points in the code.

Let's,
* add assertions in handle methods in AdventureGuideBot.java

Commit:
Delete unused methods in Ui.java and AdventureGuideBot.java

There are unused methods left in Ui.java and AdventureGuideBot.java
after updating the UI feature.

The unused methods takes additional space, the showLoadingError()
method called by AdventureGuideBot.java fails to show error text in
the main window.

Let's
* Delete unused methods in Ui.java
* Update method for handling loading error

Commit:
Extract command execution out from getResponse method

Commit:
Extract handleLoadingError() and showWelcomeMessage() out of
handleUserInput() method

Commit:
Add Streams feature of java

blacksesamezongzi committed:
Merge pull request #2 from blacksesamezongzi/branch-A-Assertions

branch-A-Assertions

Commit:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/ui/AdventureGuideBot.java

Commit:
Merge branch 'master' into branch-A-Streams

blacksesamezongzi committed:
Merge pull request #3 from blacksesamezongzi/branch-A-CodeQuality

Branch-A-code-quality

Commit:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/duke/ui/AdventureGuideBot.java

blacksesamezongzi committed:
Merge pull request #4 from blacksesamezongzi/branch-A-Streams

Add Streams feature of java

Commit:
Remove unused import statement

Commit:
Add C-tagging feature

Commit:
Add title to window

Commit:
Add a period after tagging message

Commit:
Update user guide

Commit:
Add exit feature in user guide

blacksesamezongzi committed:
Update README.md

Commit:
Update image path in README

Commit:
Add emojis in README

Commit:
Update Italic characters in README

Commit:
Add javadoc to public classes and methods

Commit:
Remove println statement used for debugging

Commit:
Update JUnit test cases

Commit:
Merge branch 'branch-A-Varargs'

# Conflicts:
#	build.gradle
#	src/main/java/duke/ui/MainWindow.java
